scripts/origin.js

```diff
@@ -0,0 +1,42 @@
+---
+layout: null
+---
+(function() {
+    function addOrUpdateQueryParam(url, param, value) {
+        if (!url || typeof url !== 'string') return url;
+        
+        var urlObj = new URL(url, window.location.origin);
```

Comment:
We may need to check for `URL` and bail if not available

We don't really have a specific min browser support here but should probably avoid errors if we can